fe/fe-core/src/main/java/com/starrocks/external/iceberg/cost/IcebergMetaCache.java

```diff
@@ -0,0 +1,118 @@
+import org.apache.iceberg.Table;
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.apache.iceberg.types.Types;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Executor;
+
+import static com.google.common.cache.CacheLoader.asyncReloading;
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+public class IcebergMetaCache {
+    private static final Logger LOG = LogManager.getLogger(IcebergMetaCache.class);
+
+    private static final long MAX_TABLE_CACHE_SIZE = 1000L;
+
+    // statistic cache
+    // IcebergTableKey => IcebergFileStats
+    @VisibleForTesting
+    protected LoadingCache<IcebergTableKey, IcebergFileStats> tableStatsCache;
+
+    public IcebergMetaCache(Executor executor) {
+        CacheBuilder<Object, Object> cacheBuilder = CacheBuilder.newBuilder();
+        cacheBuilder.expireAfterWrite(Config.iceberg_meta_cache_ttl_s, SECONDS);
+        cacheBuilder.maximumSize(MAX_TABLE_CACHE_SIZE);
+        tableStatsCache = cacheBuilder.build(asyncReloading(
+                new CacheLoader<IcebergTableKey, IcebergFileStats>() {
+                    @Override
+                    public IcebergFileStats load(IcebergTableKey key) throws Exception {
+                        Table table = getTable(key);
+                        return getIcebergFileStats(table, getCalculator(table));
+                    }
+                }, executor));
+    }
+
+    public Statistics getTableStatistics(IcebergTableKey key,
+                                         Map<ColumnRefOperator, Column> colRefToColumnMetaMap) {
+        IcebergFileStats icebergFileStats;
+        try {
+            icebergFileStats = tableStatsCache.get(key);
+        } catch (ExecutionException e) {
+            LOG.warn("Get iceberg file stats from cache with failure!", e);
+            Table table = getTable(key);
+            icebergFileStats = getIcebergFileStats(table, getCalculator(table));
+        }
+        Table table = getTable(key);
+        IcebergTableStatisticCalculator calculator = getCalculator(table);
+        List<Types.NestedField> columns = table.schema().columns();
+        return calculator.makeTableStatistics(icebergFileStats, columns, colRefToColumnMetaMap);
+    }
+
+    public List<ColumnStatistic> getColumnStatistics(IcebergTableKey key,
+                                                     Map<ColumnRefOperator, Column> colRefToColumnMetaMap) {
+        IcebergFileStats icebergFileStats;
+        try {
+            icebergFileStats = tableStatsCache.get(key);
+        } catch (ExecutionException e) {
+            LOG.warn("Get iceberg file stats from cache with failure!", e);
+            Table table = getTable(key);
+            icebergFileStats = getIcebergFileStats(table, getCalculator(table));
+        }
+        Table table = getTable(key);
```

Comment:
why not use IcebergTable.getIcebergTable() to avoid load more then once ?
at  `fe/fe-core/src/main/java/com/starrocks/external/iceberg/IcebergUtil.java:171`